### PR TITLE
Better way to detect staff-maintained packages

### PR DESCRIPTION
**Pull request by maelle**
I updated the end of the README of our private whereiseverything repo. 

If you are Scott or Jeroen, your packages developed under an rOpenSci organization will be considered staff-maintained. If you are someone else, you need to add the line ".ropensci-staff" in .Rbuildignore as suggested by @noamross.

Hopefully this line won't create Rbuildignore-related bugs.

### CODE REVIEW

File: R/registry.R [hunk 1]
```diff
@@ -163,7 +163,30 @@ get_cran_archived <- function() {
...
+  paths <- fs::path(...)
+  for (path in paths) {
+    if (fs::file_exists(path))
+      return(path)
+  }
+
+  NULL
+}
+
+  path <- path_first_existing(paste0(dir(folder, full.names = TRUE), "/", pkg_name))
```
Comment (maelle) [inline R/registry.R:180]: it's a bit stupid but at this point of the pipeline we have the codemeta.json with no trace of the folder where the package was, so we have to re-find it.

File: R/registry.R [hunk 2]
```diff
@@ -184,7 +207,7 @@ get_type <- function(status) {
 #' @importFrom ghql GraphqlClient Query
 #' @importFrom crul HttpClient
 #' @importFrom readr read_csv
-create_registry <- function(cm, outpat, time = Sys.time()) {
+create_registry <- function(cm, outpat, time = Sys.time(), folder = "repos") {
```
Comment (maelle) [inline R/registry.R:210]: argument needed for testing.

File: inst/scripts/staff.csv [hunk 3]
```diff
@@ -1,7 +1,2 @@
-Maëlle Salmon
```
Comment (maelle) [inline inst/scripts/staff.csv:1]: Only packages by Jeroen and Scott are considered staff-maintained by default.

File: tests/testthat/_snaps/create_codemeta/cm2.json [hunk 4]
```diff
@@ -160,9 +160,9 @@
   {
     "@context": ["https://doi.org/10.5063/schema/codemeta-2.0", "http://schema.org"],
     "@type": "SoftwareSourceCode",
-    "identifier": "pkg1",
+    "identifier": "pkg2",
```
Comment (maelle) [inline tests/testthat/_snaps/create_codemeta/cm2.json:163]: Most of these changes stem from the fact that packages are now named liked the folder...

**Comment by jeroen** [comment]
I suppose the part to read the `.Rbuildignore` could be written a bit more clear, perhaps using grep. Maybe somehting like this?


```r
is_staff <- maintainer %in% staff
rbuildignore <- file.path(path, ".Rbuildignore")
if(!is_staff && file.exists(rbuildignore)){
  is_staff <- any(grepl("^.ropensci-staff$", readLines(rbuildignore)))
}
```

**Pull request merged**